it/src/main/java/org/corfudb/universe/node/server/process/ProcessCorfuServer.java

```diff
@@ -0,0 +1,216 @@
+     * b) Run that jar file using java on the VM
+     */
+    @Override
+    public CorfuServer deploy() {
+        executeCommand(Optional.empty(), processManager.createServerDirCommand());
+        executeCommand(Optional.empty(), processManager.createStreamLogDirCommand());
+
+        commandHelper.copyFile(
+                params.getInfrastructureJar(),
+                processManager.getServerJar()
+        );
+        start();
+        return this;
+    }
+
+    /**
+     * Symmetrically disconnect the server from the cluster,
+     * which creates a complete partition.
+     */
+    @Override
+    public void disconnect() {
+        throw new UnsupportedOperationException("Not supported");
+    }
+
+    /**
+     * Symmetrically disconnect a server from a list of other servers,
+     * which creates a partial partition.
+     *
+     * @param servers List of servers to disconnect from
+     */
+    @Override
+    public void disconnect(List<CorfuServer> servers) {
+        throw new UnsupportedOperationException("Not supported");
+    }
+
+    /**
+     * Pause the {@link CorfuServer} process on the VM
+     */
+    @Override
+    public void pause() {
+        log.info("Pausing the Corfu server: {}", params.getName());
+
+        executeCommand(Optional.empty(), processManager.pauseCommand());
+    }
+
+    /**
+     * Start a {@link CorfuServer} process on the VM
+     */
+    @Override
+    public void start() {
+        executeCommand(
+                Optional.of(processManager.getCorfuDir()),
+                processManager.startCommand(getCommandLineParams())
+        );
+    }
+
+    /**
+     * Restart the {@link CorfuServer} process on the VM
+     */
+    @Override
+    public void restart() {
+        stop(params.getStopTimeout());
+        start();
+    }
+
+    /**
+     * Reconnect a server to the cluster
+     */
+    @Override
+    public void reconnect() {
+        throw new UnsupportedOperationException("Not supported");
+    }
+
+    /**
+     * Reconnect a server to a list of servers.
+     */
+    @Override
+    public void reconnect(List<CorfuServer> servers) {
+        throw new UnsupportedOperationException("Not supported");
+    }
+
+    /**
+     * Resume a {@link CorfuServer}
+     */
+    @Override
+    public void resume() {
+        log.info("Resuming the corfu server: {}", params.getName());
+
+        executeCommand(Optional.empty(), processManager.resumeCommand());
+    }
+
+    /**
+     * Executes a certain command on the VM.
+     */
+    private void executeCommand(Optional<Path> workDir, String cmdLine) {
+        try {
+            commandHelper.executeCommand(workDir, cmdLine);
+        } catch (IOException e) {
+            throw new NodeException("Execution error. Cmd: " + cmdLine, e);
+        }
+    }
+
+    /**
+     * @return the IpAddress of this VM.
+     */
+    @Override
+    public String getIpAddress() {
+        return "127.0.0.1";
```

Comment:
Why are we returning localhost? Also, make this a constant?

Reply:
It is an IP address, not a dns name.
the constant has extracted.